process/smartContract/process.go

```diff
@@ -49,94 +50,135 @@ type scProcessor struct {
-
-	if check.IfNil(vmContainer) {
+func NewSmartContractProcessor(args ArgsNewSmartContractProcessor) (*scProcessor, error) {
+
+	if check.IfNil(args.VmContainer) {
 		return nil, process.ErrNoVM
 	}
-	if check.IfNil(argsParser) {
+	if check.IfNil(args.ArgsParser) {
 		return nil, process.ErrNilArgumentParser
 	}
-	if check.IfNil(hasher) {
+	if check.IfNil(args.Hasher) {
 		return nil, process.ErrNilHasher
 	}
-	if check.IfNil(marshalizer) {
+	if check.IfNil(args.Marshalizer) {
 		return nil, process.ErrNilMarshalizer
 	}
-	if check.IfNil(accountsDB) {
+	if check.IfNil(args.AccountsDB) {
 		return nil, process.ErrNilAccountsAdapter
 	}
-	if check.IfNil(tempAccounts) {
+	if check.IfNil(args.TempAccounts) {
 		return nil, process.ErrNilTemporaryAccountsHandler
 	}
-	if check.IfNil(adrConv) {
+	if check.IfNil(args.AdrConv) {
 		return nil, process.ErrNilAddressConverter
 	}
-	if check.IfNil(coordinator) {
+	if check.IfNil(args.Coordinator) {
 		return nil, process.ErrNilShardCoordinator
 	}
-	if check.IfNil(scrForwarder) {
+	if check.IfNil(args.ScrForwarder) {
 		return nil, process.ErrNilIntermediateTransactionHandler
 	}
-	if check.IfNil(txFeeHandler) {
+	if check.IfNil(args.TxFeeHandler) {
 		return nil, process.ErrNilUnsignedTxHandler
 	}
-	if check.IfNil(economicsFee) {
+	if check.IfNil(args.EconomicsFee) {
 		return nil, process.ErrNilEconomicsFeeHandler
 	}
-	if check.IfNil(txTypeHandler) {
+	if check.IfNil(args.TxTypeHandler) {
 		return nil, process.ErrNilTxTypeHandler
 	}
-	if check.IfNil(gasHandler) {
+	if check.IfNil(args.GasHandler) {
 		return nil, process.ErrNilGasHandler
 	}
 
 	sc := &scProcessor{
-		vmContainer:      vmContainer,
-		argsParser:       argsParser,
-		hasher:           hasher,
-		marshalizer:      marshalizer,
-		accounts:         accountsDB,
-		tempAccounts:     tempAccounts,
-		adrConv:          adrConv,
-		shardCoordinator: coordinator,
-		scrForwarder:     scrForwarder,
-		txFeeHandler:     txFeeHandler,
-		economicsFee:     economicsFee,
-		txTypeHandler:    txTypeHandler,
-		gasHandler:       gasHandler,
-	}
-
-	err := sc.createBuiltInFunctions()
+		vmContainer:      args.VmContainer,
+		argsParser:       args.ArgsParser,
+		hasher:           args.Hasher,
+		marshalizer:      args.Marshalizer,
+		accounts:         args.AccountsDB,
+		tempAccounts:     args.TempAccounts,
+		adrConv:          args.AdrConv,
+		shardCoordinator: args.Coordinator,
+		scrForwarder:     args.ScrForwarder,
+		txFeeHandler:     args.TxFeeHandler,
+		economicsFee:     args.EconomicsFee,
+		txTypeHandler:    args.TxTypeHandler,
+		gasHandler:       args.GasHandler,
+	}
+
+	err := sc.createGasConfig(args.GasMap)
+	if err != nil {
+		return nil, err
+	}
+
+	err = sc.createBuiltInFunctions()
 	if err != nil {
 		return nil, err
 	}
 
 	return sc, nil
 }
 
+func (sc *scProcessor) createGasConfig(gasMap map[string]map[string]uint64) error {
+	baseOps := &BaseOperationCost{}
+	err := mapstructure.Decode(gasMap["BaseOperationCost"], baseOps)
```

Comment:
might have some consts for BaseOperationCost and BuiltInCost

Reply:
done.